packages/vite/src/node/server/transformRequest.ts

```diff
@@ -348,6 +348,10 @@ async function loadAndTransform(
     else mod.transformResult = result
   }
 
+  if (result && result.map) {
+    result.map.toUrl = genSourceMapUrl.bind(null, result.map);
+  }
```

Comment:
Looks like we also need to check `'version' in result.map` to narrow down the types to fix the build issue.

Also could we do `result.map.toUrl = () => genSourceMapUrl(result.map)` here so it's consistent with the importAnalysis plugin? If the `.bind()` version has a slight perf benefit, I think we can also update that in the importAnalysis plugin together.